### PR TITLE
Room Tile context menu, notifications, indicator and placement

### PR DESCRIPTION
**Pull request by t3chguy**
Fixes https://github.com/vector-im/riot-web/issues/14226
Fixes https://github.com/vector-im/riot-web/issues/14255
Fixes https://github.com/vector-im/riot-web/issues/13961

TODO:

- [x] Fix content of Notifications context menu
- [x] Wire up Notifications context menu
- [x] Use correct bell SVG

+ Moves context menus to the right places (aligns icons)
+ Adds hover states to their interactive elements
+ Flattens DOM to simplify later A11Y work
+ Simplifies some padding to match Figma (some were changed to make the hover state logical)
+ Adds and wires up Notifications context menu on Room Tile

image
image
image

### CODE REVIEW

**Comment by turt2live** [comment]
For https://github.com/vector-im/riot-web/issues/13635

File: res/css/views/rooms/_RoomTile2.scss [hunk 1]
```diff
@@ -92,20 +92,17 @@ limitations under the License.
         justify-content: center;
     }
 
-    // The menu button is hidden by default
```
Comment (turt2live) [inline res/css/views/rooms/_RoomTile2.scss:95]: this line should stay

File: src/components/structures/UserMenu.tsx [hunk 2]
```diff
@@ -191,20 +191,18 @@ export default class UserMenu extends React.Component<IProps, IState> {
...
+                </AccessibleButton>
             );
         }
 
         const elementRect = this.buttonRef.current.getBoundingClientRect();
         return (
             <ContextMenu
                 chevronFace="none"
-                left={elementRect.width + elementRect.left}
+                left={elementRect.width + elementRect.left - 20}
```
Comment (turt2live) [inline src/components/structures/UserMenu.tsx:205]: A comment for the 20 would be nice

File: src/components/views/rooms/RoomTile2.tsx [hunk 3]
```diff
@@ -153,55 +203,120 @@ export default class RoomTile2 extends React.Component<IProps, IState> {
         this.setState({generalMenuDisplayed: false}); // hide the menu
     };
 
+    private async saveNotifState(ev: ButtonEvent, newState: ALL_MESSAGES_LOUD | ALL_MESSAGES | MENTIONS_ONLY | MUTE) {
+        ev.preventDefault();
+        ev.stopPropagation();
+        if (MatrixClientPeg.get().isGuest()) return;
+
+        try {
+            // TODO add local echo
```
Comment (turt2live) [inline src/components/views/rooms/RoomTile2.tsx:212]: this and the following TODO comment should either be resolved before landing or link to issues we can prioritize

File: src/components/views/rooms/RoomTile2.tsx [hunk 4]
```diff
@@ -153,55 +203,120 @@ export default class RoomTile2 extends React.Component<IProps, IState> {
...
+            await setRoomNotifsState(this.props.room.roomId, newState);
+        } catch (error) {
+            // TODO: some form of error notification to the user to inform them that their state change failed.
```
Comment (t3chguy) [inline src/components/views/rooms/RoomTile2.tsx:215]: This is an existing TODO in the current RoomTileContextMenu

**Pull request merged**